src/widgets/JudgePanel.ts

```diff
@@ -334,7 +333,7 @@ export class JudgePanel extends BoxPanel {
 
     const results: IRunResult[] = [];
     for (const testCase of testCases) {
-      const result = await this.runWithInput(kernel, problem, testCase);
+      const result = await this.runWithInput(this.session, problem, testCase);
```

Comment:
you should pass `sessionContext` not `this.session`. We use seperate session context for judging!